ssl/statem/extensions_clnt.c

```diff
@@ -16,10 +16,26 @@ EXT_RETURN tls_construct_ctos_renegotiate(SSL_CONNECTION *s, WPACKET *pkt,
                                           unsigned int context, X509 *x,
                                           size_t chainidx)
 {
-    /* Add RI if renegotiating */
-    if (!s->renegotiate)
-        return EXT_RETURN_NOT_SENT;
+    if (!s->renegotiate) {
+        /* If not renegotiating, send an empty RI extension to indicate support */
+
+        if (s->min_proto_version <= TLS1_VERSION) {
```

Comment:
Should this be `s->min_proto_version <= TLS1_VERSION || s->min_proto_version >= TLS1_3_VERSION` ? However maybe even `!SSL_CONNECTION_IS_DTLS(s) && (s->min_proto_version <= TLS1_VERSION || s->min_proto_version >= TLS1_3_VERSION)` would be more correct along with changing the condition below.

Reply:
> Is that enough? Happy to double-check that here regardless, if that's helpful.

I am afraid it is not enough as indicated by the trace ref test on QUIC as that should be using TLS-1.3 only but it still adds the empty extension.

Reply:
Ah, understood! I had misinterpreted that test trace entirely.

Trace now reverted, so it's unmodified by this change, and I've added the TLS 1.3 check you suggested here to skip this extension in that case.